### PR TITLE
reduce use of @overload in socket stubs

### PR DESCRIPTION
**Pull request by Naruto0**
fixing https://github.com/python/mypy/issues/430

### CODE REVIEW

**Comment by gvanrossum** [comment]
Thanks! Much cleaner. Can you do the same for the 2.7 version?

**Comment by Naruto0** [comment]
Done.

File: stdlib/2.7/socket.pyi [hunk 1]
```diff
@@ -283,36 +283,17 @@ class socket:
...
     def fileno(self) -> int: ...
 
     # return value is an address
     def getpeername(self) -> Any: ...
     def getsockname(self) -> Any: ...
-
-    @overload
-    def getsockopt(self, level: int, optname: str) -> str: ...
-    @overload
     def getsockopt(self, level: int, optname: str, buflen: int) -> str: ...
```
Comment (gvanrossum) [inline stdlib/2.7/socket.pyi]: Actually this needs to keep the overload. But the original was also wrong: return type for `getsockopt(int, str)` is int, not str.

File: stdlib/3/socket.pyi [hunk 2]
```diff
@@ -282,36 +282,17 @@ class socket:
...
     def fileno(self) -> int: ...
 
     # return value is an address
     def getpeername(self) -> Any: ...
     def getsockname(self) -> Any: ...
-
-    @overload
-    def getsockopt(self, level: int, optname: str) -> bytes: ...
-    @overload
     def getsockopt(self, level: int, optname: str, buflen: int) -> bytes: ...
```
Comment (gvanrossum) [inline stdlib/3/socket.pyi]: Again, keep the overload but make the first one return int.

**Comment by Naruto0** [comment]
Thank you very much. I was about to ask if this case is somehow different, but forgot to. I'll fix it as soon as possible,

**Comment by gvanrossum** [comment]
Thanks! Ready to merge.

**Pull request merged**